checks/shell_download_validate.go

```diff
@@ -0,0 +1,628 @@
+		if len(cmd) >= 3 && strings.EqualFold("s3api", cmd[1]) && strings.EqualFold("get-object", cmd[2]) {
+			return true
+		}
+	}
+	return false
+}
+
+func getWgetOututFile(cmd []string) (pathfn string, ok bool, err error) {
+	if isBinaryName("wget", cmd[0]) {
+		for i := 1; i < len(cmd)-1; i++ {
+			// Find -O output, or use the basename from url.
+			if strings.EqualFold(cmd[i], "-O") {
+				return cmd[i+1], true, nil
+			}
+		}
+
+		// Could not find -O option, use the url's name instead.
+		for i := 1; i < len(cmd); i++ {
+			if !strings.HasPrefix(cmd[i], "http") {
+				continue
+			}
+
+			u, err := url.Parse(cmd[i])
+			if err != nil {
+				return "", false, fmt.Errorf("url.Parse: %w", err)
+			}
+			return path.Base(u.Path), true, nil
+		}
+	}
+	return "", false, nil
+}
+
+func getGsutilOututFile(cmd []string) (pathfn string, ok bool, err error) {
+	if isBinaryName("gsutil", cmd[0]) {
+		for i := 1; i < len(cmd)-1; i++ {
+			if !strings.HasPrefix(cmd[i], "gs://") {
+				continue
+			}
+			pathfn := cmd[i+1]
+			if filepath.Clean(filepath.Dir(pathfn)) == filepath.Clean(pathfn) {
+				// Directory.
+				u, err := url.Parse(cmd[i])
+				if err != nil {
+					return "", false, fmt.Errorf("url.Parse: %w", err)
+				}
+				return filepath.Join(filepath.Dir(pathfn), path.Base(u.Path)), true, nil
+			}
+
+			// File provided.
+			return pathfn, true, nil
+		}
+	}
+	return "", false, nil
+}
+
+func getAwsOututFile(cmd []string) (pathfn string, ok bool, err error) {
+	// https://docs.aws.amazon.com/AmazonS3/latest/userguide/download-objects.html.
+	if isBinaryName("aws", cmd[0]) {
+		if len(cmd) < 3 || !strings.EqualFold("s3api", cmd[1]) || !strings.EqualFold("get-object", cmd[2]) {
+			return "", false, nil
+		}
+
+		// Just take the last 2 arguments.
+		ifile := cmd[len(cmd)-2]
+		ofile := cmd[len(cmd)-1]
+		if filepath.Clean(filepath.Dir(ofile)) == filepath.Clean(ofile) {
+			u, err := url.Parse(ifile)
+			if err != nil {
+				return "", false, fmt.Errorf("url.Parse: %w", err)
+			}
+			return filepath.Join(filepath.Dir(ofile), path.Base(u.Path)), true, nil
+		}
+
+		// File provided.
+		return ofile, true, nil
+	}
+	return "", false, nil
+}
+
+func getOutputFile(cmd []string) (pathfn string, ok bool, err error) {
+	if len(cmd) == 0 {
+		return "", false, nil
+	}
+
+	// Wget.
+	fn, b, err := getWgetOututFile(cmd)
+	if err != nil || b {
+		return fn, b, err
+	}
+
+	// Gsutil.
+	fn, b, err = getGsutilOututFile(cmd)
+	if err != nil || b {
+		return fn, b, err
+	}
+
+	// Aws.
+	fn, b, err = getAwsOututFile(cmd)
+	if err != nil || b {
+		return fn, b, err
+	}
+
+	return "", false, nil
+}
+
+func isInterpreter(cmd []string) bool {
+	if len(cmd) == 0 {
+		return false
+	}
+
+	for _, b := range interpreters {
+		if isBinaryName(b, cmd[0]) {
+			return true
+		}
+	}
+	return false
+}
+
+func isInterpreterWithFile(cmd []string, fn string) bool {
+	if len(cmd) == 0 {
+		return false
+	}
+
+	for _, b := range interpreters {
+		if isBinaryName(b, cmd[0]) {
```

Comment:
nit: avoid extra indentation for lines 188- by inverting this condition and continuing

Reply:
done.